src/pluggy/manager.py

```diff
@@ -118,10 +118,11 @@ def register(self, plugin, name=None):
                 normalize_hookimpl_opts(hookimpl_opts)
                 method = getattr(plugin, name)
                 hookimpl = HookImpl(plugin, plugin_name, method, hookimpl_opts)
-                hook = getattr(self.hook, name, None)
+                specname = hookimpl_opts.get("specname") or name
```

Comment:
Is `hookimpl_opts.get("specname", name)` maybe more standard/efficient?

I also wonder if maybe you should keep the local variable name as `name` since the following 3 lines then won't need to be changed.

So instead do `name = hookimpl_opts.get("specname", name)`

Reply:
the only difference I can see with `hookimpl_opts.get("specname", name)` vs `or name` is the case when `specname == ''`  (`or` will fall back to name, but the `get` method wouldn't).  but maybe if someone has gone through the effort to explicitly make `specname` the empty string, we should just let them?

will return local var -> `name`

Reply:
oh that's right, there _was_ a reason:  providing a default to `get` also doesn't work if `specname=None` ... which is the default value.
```python
hookimpl_opts = {'specname': None}
hookimpl_opts.get("specname", name)  # returns None
hookimpl_opts.get("specname") or name  # returns name
```

Reply:
Ah touche; good point.